tests/unit/amo/pages/TestHome.js

```diff
@@ -319,6 +319,38 @@ describe(__filename, () => {
     );
   });
 
+  it('dispatches an action to fetch the add-ons to display on update', () => {
+    const includeFeaturedThemes = false;
+    const includeTrendingExtensions = false;
+    const errorHandler = createStubErrorHandler();
+    const { store } = dispatchClientMetadata();
+
+    const fakeDispatch = sinon.stub(store, 'dispatch');
+
+    const root = render({
+      errorHandler,
+      includeFeaturedThemes,
+      includeTrendingExtensions,
+      store,
+    });
+    fakeDispatch.resetHistory();
+
+    // We simulate an update to trigger `componentDidUpdate()`.
+    root.setProps();
+
+    sinon.assert.callCount(fakeDispatch, 2);
+    sinon.assert.calledWith(fakeDispatch, setViewContext(VIEW_CONTEXT_HOME));
+    sinon.assert.calledWith(
+      fakeDispatch,
+      fetchHomeAddons({
+        errorHandlerId: errorHandler.id,
```

Comment:
This would require less setup if you passed `root.instance().props.errorHandler.id`